game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java

```diff
@@ -77,23 +67,27 @@ public LobbyServerPropertiesFetcher() {
     if (fromHostedFile.isPresent()) {
       return fromHostedFile;
     }
-
     return getLastUsedProperties();
   }
 
   private static Optional<LobbyServerProperties> getTestOverrideProperties() {
-    return getTestOverrideProperties(ClientSetting.testLobbyHost, ClientSetting.testLobbyPort);
+    return getTestOverrideProperties(
+        ClientSetting.testLobbyHost,
+        ClientSetting.testLobbyPort,
+        ClientSetting.httpLobbyUriOverride);
   }
 
   @VisibleForTesting
   static Optional<LobbyServerProperties> getTestOverrideProperties(
       final GameSetting<String> testLobbyHostSetting,
-      final GameSetting<Integer> testLobbyPortSetting) {
-    if (testLobbyHostSetting.isSet() && testLobbyPortSetting.isSet()) {
+      final GameSetting<Integer> testLobbyPortSetting,
+      final GameSetting<String> testLobbyHttpUri) {
+    if (testLobbyHostSetting.isSet() && testLobbyPortSetting.isSet() && testLobbyHttpUri.isSet()) {
```

Comment:
I'm still of the opinion that if one of these is set--but not all--we should log it.  A dev may not realize that they have to set them _all_ in order for _any_ of them to be considered.  Alternatively, the engine preferences UI could be updated to enforce this.

Reply:
Agree, IMO it has magnified an existing problem. On the plus side it is generally obvious when the wrong lobby is chosen.

IMO Enforcing at UI level is ideal. In addition to that, I think we could also benefit to decouple the datatypes between java lobby and  http lobby data.  Last, I think it'd be a good move to combine lobby host and port to a URI so that it is a single value and not two.